src/lib/snark_params/scan_state_constants.ml

```diff
@@ -0,0 +1,46 @@
+[%%import
+"../../config.mlh"]
+
+(* in the config files, we either have:
+
+    - transaction_capacity_log_2 and work_delay exp explicitly, or
+    - those are calculated from the tps_goal and num_snark_workers
+
+   use the config boolean scan_state_calc_params to distinguish those cases
+
+ *)
+
+module type S = Scan_state_constants_intf.S
+
+[%%if
+scan_state_calc_params]
+
+open Core_kernel
+
+[%%inject
+"tps_goal", scan_state_tps_goal]
+
+[%%inject
+"num_snark_workers", scan_state_num_snark_workers]
+
+[%%inject
+"block_window_duration", block_window_duration]
+
+let max_coinbases = 2
+
+let max_user_commands_per_block = tps_goal * block_window_duration
+
+let transaction_capacity_log_2 =
+  1 + Int.ceil_log2 (max_user_commands_per_block + max_coinbases)
+
+let work_delay = Int.pow 2 transaction_capacity_log_2 / num_snark_workers
```

Comment:
What's the justification for this calculation?

Reply:
Number of snark workers here assumes the proving time for a work bundle is the same as `block_window_duration` which is currently the case. It takes about 5 mins for generating two snarks. Therefore, you'd get 1 snark work per block per snark worker. We should have it as a different variable for it though in case it changes